python/lsst/ip/isr/calibType.py

```diff
@@ -209,7 +215,7 @@ def updateMetadata(self, camera=None, detector=None, filterName=None,
         self._metadata["INSTRUME"] = self._instrument if self._instrument else None
         self._metadata["RAFTNAME"] = self._raftName if self._raftName else None
         self._metadata["SLOTNAME"] = self._slotName if self._slotName else None
-        self._metadata["DETECTOR"] = self._detectorId if self._detectorId else None
+        self._metadata["DETECTOR"] = self._detectorId if self._detectorId is not None else None
```

Comment:
I don't understand what this is doing. As far I can tell it is saying to set this entry to the value in the object if it it not None but to set it to None if it is None. Isn't that the same as letting it set the value directly? Since the only case where you get None is if the `self._detectorId` was itself None and therefore you get the extra logic without any gain.

The other lines are converting False to None but this one is converting None to None.

Reply:
This was a fix for the case where `self._detectorId = 0`.  `_detectorId` is now expected to be an integer, not a string, and in this case `if self._detectorId` returns false, converting the integer 0 into a None.  In turn, this was breaking round-trip tests.

Reply:
I still don't understand though. I must be missing something.

For:

```python
x = self._detectorId if self._detectorId is not None else None
```

This can be rewritten as:

```python
if self._detectorId is not None:
    x = self_detectorId
else:
    x = None
```

which is the same as:

```python
if self._detectorId is None:
    x = None
else:
    x = self._detectorId
```

which is the same as:

```python
x = self._detectorId
```
isn't it?

Reply:
I agree with all that, leading me to believe the test case failure confused me.  Replaced with the simple version.

Reply:
I'm sure you're are right that `self._detectorId if self._detectorId else None` was wrong because of the `0` so this did need fixing.